airflow/models/dag.py

```diff
@@ -752,6 +752,10 @@ def get_task_instances(
 
     @property
     def roots(self):
+        return [t for t in self.tasks if not t.upstream_list]
+
+    @property
+    def leaves(self):
```

Comment:
Can we add python docs here: `Nodes that have no children. These are called leaves or leaf nodes.`